geode-gfsh/src/main/java/org/apache/geode/management/internal/cli/commands/StopGatewaySenderOnMemberFactory.java

```diff
@@ -0,0 +1,20 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package org.apache.geode.management.internal.cli.commands;
+
+interface StopGatewaySenderOnMemberFactory {
```

Comment:
Is it really necessary to add those interfaces? Personally, I don't like adding more interfaces/implementations only for testing purposes. I thought change set was good, just missing a few more tests to cover all branches.  You can easily test your previous command changes by mocking up some dependencies.

Reply:
@kirklund I went for the `Supplier` option.
@jinmeiliao I have removed some interfaces (by using the `Supplier` one as Kirk suggested) and I have moved the ones remaining to the class using them.